celements-scheduler/src/main/java/com/celements/scheduler/job/AbstractJob.java

```diff
@@ -100,94 +91,49 @@ public final void execute(JobExecutionContext jobContext) throws JobExecutionExc
-    dummy.setHost(xwikiContext.getRequest().getHeader("x-forwarded-host"));
-    dummy.setScheme(xwikiContext.getRequest().getScheme());
-    XWikiServletRequest request = new XWikiServletRequest(dummy);
-
-    // Force forged context response to a stub response, since the current context response
-    // will not mean anything anymore when running in the scheduler's thread, and can cause
-    // errors.
-    XWikiResponse response = new XWikiServletResponseStub();
-
-    // IMPORTANT: do NOT clone xwikiContext. You would need to ensure that no reference or
-    // unwanted value leaks in the new context.
-    // IMPORTANT: following lines base on XWikiRequestInitializer.prepareContext
-    XWikiContext scontext = new XWikiContext();
-    scontext.setEngineContext(xwikiContext.getEngineContext());
-    scontext.setRequest(request);
-    scontext.setResponse(response);
-    scontext.setAction("view");
-    scontext.setDatabase(database);
-
-    // feed the job context
-    scontext.setUser(xwikiContext.getUser());
-    scontext.setLanguage(xwikiContext.getLanguage());
-    scontext.setMainXWiki(xwikiContext.getMainXWiki());
-    scontext.setMode(XWikiContext.MODE_SERVLET);
-
-    scontext.setWiki(xwiki);
-    scontext.getWiki().getStore().cleanUp(scontext);
-
-    scontext.flushClassCache();
-    scontext.flushArchiveCache();
-    return scontext;
+  private ExecutionContext createEContextForJob(JobDataMap data) {
+    ExecutionContext context = new ExecutionContext();
+    execution.get().setContext(context);
+    XWikiDocument jobDoc = ((XWikiDocument) data.get("jobDoc")).clone();
+    context.set(DOC, jobDoc);
+    context.set(WIKI, Optional.ofNullable(nullToEmpty(data.getString("jobDatabase").trim()))
+        .map(WikiReference::new)
+        .orElseGet(() -> jobDoc.getDocumentReference().getWikiReference()));
+    context.set(XWIKI_REQUEST_ACTION, "view");
+    return context;
   }
 
-  void setupServerUrlAndFactory(XWikiContext scontext, XWikiContext xwikiContext)
-      throws MalformedURLException, DocumentNotExistsException {
-    scontext.setDoc(getModelAccess().getDocument(xwikiContext.getDoc().getDocumentReference()));
-
-    final URL url = xwikiContext.getWiki().getServerURL(xwikiContext.getDatabase(), scontext);
-    // Push the URL into the slf4j MDC context so that we can display it in the generated logs
-    // using the %X{url} syntax.
-    MDC.put("url", url.toString());
-    scontext.setURL(url);
-
-    com.xpn.xwiki.web.XWikiURLFactory xurf = xwikiContext.getURLFactory();
-    if (xurf == null) {
-      xurf = scontext.getWiki().getURLFactoryService().createURLFactory(scontext.getMode(),
-          scontext);
+  /**
+   * Feed the stub xwiki context created by the ExecutionContextManager with additional job data for
+   * the job execution thread.
+   */
+  private void prepareXContextForJob(JobDataMap data) {
+    XWikiContext context = getXWikiContext();
+    context.setUser("XWiki.Scheduler");
```

Comment:
please put it in the else-statement instead of calling context.setUser twice. It is complex to understand that this is only the fallback value if "jobUser" is missing.